### PR TITLE
[GEOS-7603] Add support for dynamically choosing jpeg or png compression based on output contents

### PR DESCRIPTION
**Pull request by aaime**
Depends on:
https://github.com/GeoWebCache/geowebcache/pull/395

(might not build/work without it)

### CODE REVIEW

File: src/wms/src/main/java/org/geoserver/wms/map/MetatileMapOutputFormat.java [hunk 1]
```diff
@@ -37,6 +37,9 @@
 import org.geotools.resources.i18n.Errors;
 import org.geotools.util.logging.Logging;
 
+import it.geosolutions.jaiext.BufferedImageAdapter;
+import it.geosolutions.rendered.viewer.RenderedImageBrowser;
```
Comment (dromagnoli) [inline src/wms/src/main/java/org/geoserver/wms/map/MetatileMapOutputFormat.java]: oversight: RenderedImageBrowser has been probably imported during some interactive debug :) I would remove it.

File: src/wms/src/main/java/org/geoserver/wms/map/MetatileMapOutputFormat.java [hunk 2]
```diff
@@ -129,6 +132,7 @@ public WebMap produceMap(WMSMapContent mapContent) throws ServiceException, IOEx
 
         synchronized (key) {
             RenderedImage tile = tileCache.getTile(key, request);
+            tile = null; // REMOVE ME!
```
Comment (dromagnoli) [inline src/wms/src/main/java/org/geoserver/wms/map/MetatileMapOutputFormat.java]: REMOVE IT! :)

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetMapIntegrationTest.java [hunk 3]
```diff
@@ -1118,4 +1118,31 @@ public void testAdvancedProjectionHandling() throws Exception {
...
+        assertNotBlank("testJpegPngOpaque", image);
+    }
+    
+    @Test
+    public void testJpegPngEmpty() throws Exception {
+        String request = "wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&FORMAT=image%2Fvnd.jpeg-png&TRANSPARENT=true&STYLES"
+                + "&LAYERS=cite%3ABasicPolygons&SRS=EPSG%3A4326&WIDTH=256&HEIGHT=256&BBOX=-1.9%2C1.8%2C-1.3%2C2.5";
+        // checks it's a PNG
+        BufferedImage image = getAsImage(request, "image/png");
+        assertBlank("testJpegPngOpaque", image, new Color(255,255,255,0));
```
Comment (dromagnoli) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetMapIntegrationTest.java]: Not a big deal but you could update the message in the assertion text as "testJpegPngEmpty" to reflect the name of the test. I think it is a simple copy and paste typo.

**Comment by dromagnoli** [comment]
Beside my minor comments, it looks good to me.

File: src/wms/src/main/java/org/geoserver/wms/map/MetatileMapOutputFormat.java (hunk 2 above)
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/map/MetatileMapOutputFormat.java]: YESSIR!

**Pull request merged**